src/compiler/crystal/semantic/type_declaration_processor.cr

```diff
@@ -631,6 +624,28 @@ struct Crystal::TypeDeclarationProcessor
     nil
   end
 
+  private def remove_duplicate_instance_vars_declarations
+    remove_duplicate_instance_vars_declarations(@program)
+  end
+
+  private def remove_duplicate_instance_vars_declarations(type : Type)
```

Comment:
would this do the same thing?

```suggestion
  private def remove_duplicate_instance_vars_declarations(type : Type = @program)
```

Reply:
Yes, though I personally prefer to distinguish a main entry point and a helper method. I might change it though. Thanks!

Reply:
I pushed another commit and now it makes less sense to do it with default arguments.